lib/THC/THCReduce.cuh

```diff
@@ -11,24 +11,65 @@
 #include "THCTensorTypeUtils.cuh"
 #include "THCReduceApplyUtils.cuh"
 
-// Threads per thread block
-#define THC_NONCONTIG_REDUCE_BLOCK_SIZE 32 * 16
+#include <cstdio>
 
-template <typename IndexType>
-__device__ __forceinline__ IndexType getReduceNoncontigDimSliceIndex() {
-  // Each thread handles one slice
-  return getLinearBlockId<IndexType>() * THC_NONCONTIG_REDUCE_BLOCK_SIZE + threadIdx.x;
+#define LOCAL_MAX_BLOCK_SIZE 512
```

Comment:
https://github.com/csarofeen/cutorch/blob/master/lib/THC/THCReduce.cuh#L239-L244
Ensures block size = 512, was a little bit of a misnomer as I enforced 512 instead of having it as a max.